vars/buildPlugin.groovy

```diff
@@ -61,7 +61,14 @@ def call(Map params = [:]) {
     }
 
     tasks[stageIdentifier] = {
-      retry(count: 3, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
+      int retryCounts = 1
+      retry(count: 2, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
+        if (retryCounts == 2 && platform != 'windows') {
+          // no spot instances for windows for now TODO change when available
+          // Use a spot instance for the first try and nonspot for second
+          label = 'ubuntu-22-amd64-maven17-nonspot'
+        }
+        retryCounts = retryCounts + 1
```

Comment:
why change this from 3?
```suggestion
      int retryCount = 0
      retry(count: 3, conditions: [kubernetesAgent(handleNonKubernetes: true), nonresumable()]) {
        if (retryCount == 1 && platform != 'windows') {
          // no spot instances for windows for now TODO change when available
          // Use a spot instance for the first try and nonspot for second
          label = 'ubuntu-22-amd64-maven17-nonspot'
        }
       retryCount++
```

Reply:
only 2 retry instead of 2, it's just to match one on spot, one on non spot ...
we can also change line 66 to != 0 then nonspot and put back 3 tries ... (1 spot, 3 nonspot)

WDYT ?

Reply:
@smerle33 I missed the old value of 3. But it make sense to retry twice (with spot) and only fallback ondemand for the last one (that was the previous behavior). Tim's suggestion looks good.